### PR TITLE
:tada: Source GitHub - added more meaningful error description for connector check

### LINKED ISSUE #841: Source Github - Validate repo name entered correctly in connector config
## Tell us about the problem you're trying to solve
if I incorrectly input the repo name in the github (e.g: not specifying the organization name), creating the source connection does not fail. Instead i end up with data when I run a sync. 

## Describe the solution you’d like
I'd like creating a GH source to fail with a clear message when I input the repo or org name incorrectly.

### PR DESCRIPTION
**Pull request by midavadim**
## What
Errors in UI during connector check:

wrong repo name:
HTTPError('404 Client Error: Not Found for url: https://api.github.com/repos/airbytehq/airbyte3?per_page=100')"

wrong org name:
HTTPError('404 Client Error: Not Found for url: https://api.github.com/orgs/airbytehqBLA/repos?per_page=100')"

## How
Added meaningful description for error during connector check

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [x] Create a non-forked branch based on this PR and test the below items on it
- [x] Build is successful
- [x] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [x] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [x] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [x] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [x] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

### CODE REVIEW

File: airbyte-integrations/connectors/source-github/source_github/source.py [hunk 1]
```diff
@@ -146,8 +146,21 @@ def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) ->
             for stream_slice in repository_stats_stream.stream_slices(sync_mode=SyncMode.full_refresh):
                 next(repository_stats_stream.read_records(sync_mode=SyncMode.full_refresh, stream_slice=stream_slice), None)
             return True, None
+
         except Exception as e:
-            return False, repr(e)
+            message = repr(e)
+            if '404 Client Error: Not Found for url: https://api.github.com/repos/' in message:
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-github/source_github/source.py:152]: Can we have the `status_code` and `error message` as a separate variables and check for match based on them?
something like:
The code is just an example, check the methods for `e` based on what you have as an exception.
```
error_code = e.response.status_code
error_msg = e.response.json().get("message")

if error_code in [404, ...] and error_msg in ["https://api.github.com/repos/"]:
    ....
```

Comment (antixar) [reply airbyte-integrations/connectors/source-github/source_github/source.py:152]: yes, I guess the GitHub API should return normal error responses. We have to check this

**Review by antixar — APPROVED** [review]
LGTM
Please publish your changes

**Review by bazarnov — APPROVED** [review]

**Pull request merged**